### PR TITLE
DDF-2814 -Backport- Fixed self-intersecting polygon searches

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Backport cherry pick of https://github.com/codice/ddf/pull/1704
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@brendan-hofmann @emmberk @mcalcote
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
UI
Core APIs
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@andrewkfiedler
@pklinef
#### How should this be tested? (List steps with links to updated documentation)
Install DDF and go to the catalog ui.
Create a custom local search for anyGeo with an intersecting polygon.
Verify that the search completes without any errors.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2814

#### Checklist:
- [X] Documentation Updated
- [ ] Change Log Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlResult.java [hunk 1]
```diff
@@ -60,9 +61,13 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CqlQueryResponse.class);
 
+    // For queries we us repairConvexHull which my cause false positives to be returns but this
```
Comment (emmberk) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlResult.java]: typo: us -> use

Comment (emmberk) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlResult.java]: typo: returns -> returned

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java [hunk 2]
```diff
@@ -82,6 +92,21 @@
 
     private static final WKTWriter WKT_WRITER = new WKTWriter();
 
+    // For queries we us repairConvexHull which my cause false positives to be returns but this
```
Comment (emmberk) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java]: typo: us -> use

Comment (emmberk) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java]: typo: returns -> returned

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlResult.java (hunk 1 above)
Comment (clockard) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlResult.java]: Oops. Fixed

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java (hunk 2 above)
Comment (clockard) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java]: Fixed

**Pull request merged**